### ISSUE TITLE
d2bot.exe crashes

### ISSUE THREAD

**Issue by spark10z** [kolton/d2bot-with-kolbot#653]
hi,

i am running kolbot on 3 different computer.  all are windows 10, updated to current version.
kolbot on 2 pc are running without problem mostly, but on one pc, i got repeatedly sudden d2bot.exe crashes. i run 4-7 chars and one of them is static mule char keeping game in act 1 town ( i wrote lil script ).
other chars are doing mf run of 20-25 min a game.

it seems crash happens when one of mf char try to call mule in chat lobby, or at least one of char is in lobby before creating next game.  just d2bot.exe crashes and mf char will continue mf run without problem, just they cant call mule anymore naturally.
d2bot.exe is running as admin mode, and using webroot anti virus.

any advice ?  or can you give me link to useful information ?   pls help me.
thank you

**Comment by noah-** [comment]
is there anything in /logs/exceptions.log?

**Comment by spark10z** [comment]
thank you for asking. because of sudden crash, there was no log line in exception log.

i have tried as many points as i can think of, and it seems now the sudden crash problem has gone.
what i did are...
1.  sfc /scannow
2. ccleaner with d2 related log, cache files included as addition
3. etal's d2maintenace adopted to kolbot
4. glary utility to clean all
5. jv-powertool's registry cleaner
6. erased huge log files in subfolders in trunk -> all contents cleared and file size are now zero

i am strongly doubting huge log files caused of crash problem. i ll see for a while if it will not happen again, and will report.

**Comment by spark10z** [comment]
Since i performed maintenance as in last post, it seems now d2bot.exe is running stable.

only problem i sometimes receive is "d2bot.exe not resonding" error.  in some cases, this message is printed to console for many times.. like 30 lines, then suddenly stopping to print error message and bot returns to normal mf runs.   

Or, other case is way serious, after some mf runs, my char try to change key.  after relog in to account, he stands in lobby and not making next game anymore.  in console's status shows <running...> and will not proceed to next step.

in second case, i need to stop all char games, and restart d2bot.exe, then they can run normally for some hours. 

anyone got any suggestion ?    please advice me :)     thank you

**Comment by iihieiii** [comment]
### Hi, i have made some changes and they seems to have fixed the problem: 


:arrow_right: **1)** FIRST please Check if ***D2bot.exe*** and ***Game.exe*** TRULY are in ***"Executable as Admin" and "Compatiblity mode with WIN XP SERVICE PACK 2"*** (mine was on 3), and also apply this for all user on : Properties/Security tab/Edit, and then check if all user can do this ;)

### I know all this things seems **_superstitious_** BUT after all this changes im runnig the bot for 2 hours and had only 1 crash lol, way better than the crash every minute lol.


***---- NOW THE REAL CHANGES ------***

:arrow_right: **2)** Check if the folders on: 
- d2bot-with-kolbot-master\d2bs\kolbot\libs\ ***config***
- d2bot-with-kolbot-master\d2bs\kolbot\ ***pickit***

Have no other Files or Folders beyond the originals ( mine had alot folders that i made like little ''backups'' files for edeting kolton.nip), if you have alot trash files like me, just place them to another location in you desktop.

:arrow_right: **3)** In the folder : ***d2bot-with-kolbot-master\d2bs\kolbot\data***, if you have changed the files to restart runs counting ( like i did sometimes lol), i recommend you to make a new Profile and delete the older one ;)

:arrow_right: **4)** Set ***Config.ItemInfoQuality = [ ];***, on your "char name".js file, has no numbers at all (like this [ ]; ) because i had put some numbers there but i forgot to remove them :(

:arrow_right: **5)** Set ***Config.LogExperience = false;*** , for some reason i sense that this makes more crashes when is set to True.

:arrow_right: **6)** If you are runing the ***Scripts.Cows = true;*** , try to put this script has the last script to run, just after the last Baal script.

:arrow_right: **7)** If you use this line in your kolton.nip for keys : ***( [name] == key # # [maxquantity] == 12)*** , try to put this line in the end of your kolton.nip file, has the last line of command.

:arrow_right: **8)** Also set the Profile on the bot to run with : ***-w*** and ***-ns***

:arrow_right: **9)** And the last but not least, ***DON`T RUN the d2bot.exe FROM A SHORTCUT*** (even if its from your desktop! )
Open your folder **d2bot-with-kolbot-master** and double-click d2bot.exe from there! ;)


***I know all this things seems **_superstitious_** BUT after all this changes im runnig the bot for 2 hours and had only 1 crash lol, way better than the crash every minute lol.***

![Image of kolbot no crashs](https://i.imgur.com/2X2wFId.gif)

:heavy_check_mark: ***Please try it out ;)*** :heavy_check_mark:

**Comment by bobzen1234** [comment]
> ### Hi, i have made some changes and they seems to have fixed the problem:
> ➡️ **1)** FIRST please Check if _**D2bot.exe**_ and _**Game.exe**_ TRULY are in _**"Executable as Admin" and "Compatiblity mode with WIN XP SERVICE PACK 2"**_ (mine was on 3), and also apply this for all user on Compatibility Tab for both .exe's.
> 
> ### I know all this things seems **_superstitious_** BUT after all this changes im runnig the bot for 2 hours and had only 1 crash lol, way better than the crash every minute lol.
> _**---- NOW THE REAL CHANGES ------**_
> 
> ➡️ **2)** Check if the folders on:
> 
> * d2bot-with-kolbot-master\d2bs\kolbot\libs\ _**config**_
> * d2bot-with-kolbot-master\d2bs\kolbot\ _**pickit**_
> 
> Have no other Files or Folders beyond the originals ( mine had alot folders that i made like little ''backups'' files for edeting kolton.nip), if you have alot trash files like me, just place them to another location in you desktop.
> 
> ➡️ **3)** In the folder : _**d2bot-with-kolbot-master\d2bs\kolbot\data**_, if you have changed the files to restart runs counting ( like i did sometimes lol), i recommend you to make a new Profile and delete the older one ;)
> 
> ➡️ **4)** Set _**Config.ItemInfoQuality = [ ];**_, on your "char name".js file, has no numbers at all (like this [ ]; ) because i had put some numbers there but i forgot to remove them :(
> 
> ➡️ **5)** Set _**Config.LogExperience = false;**_ , for some reason i sense that this makes more crashes when is set to True.
> 
> ➡️ **6)** If you are runing the _**Scripts.Cows = true;**_ , try to put this script has the last script to run, just after the last Baal script.
> 
> ➡️ **7)** If you use this line in your kolton.nip for keys : _**( [name] == key # # [maxquantity] == 12)**_ , try to put this line in the end of your kolton.nip file, has the last line of command.
> 
> ➡️ **8)** Also set the Profile on the bot to run with : _**-w**_ and _**-ns**_
> 
> ➡️ **9)** And the last but not least, _**DON`T RUN the d2bot.exe FROM A SHORTCUT**_ (even if its from your desktop! )
> Open your folder **d2bot-with-kolbot-master** and double-click d2bot.exe from there! ;)
> 
> _**I know all this things seems **_superstitious_** BUT after all this changes im runnig the bot for 2 hours and had only 1 crash lol, way better than the crash every minute lol.**_
> 
> ![Image of kolbot no crashs](https://camo.githubusercontent.com/4ccf28ae3051ac1cbc10e58751ba1c9386248745/68747470733a2f2f692e696d6775722e636f6d2f325832774649642e676966)
> 
> ✔️ _**Please try it out ;)**_ ✔️

This did not work for me. It crashed everytime it launched before I had 10 crashes in 400 games. Tried to change mine to windows 7 and see if it gets better :) hopefully 0 crashes in 400 games now :b

**Comment by iihieiii** [comment]
yeah....im sorry....didnt work for me either...ate least for a short peirodo of time....i will try to reinstall everything and update my windows... ;) byebye
